lib/hive/controller/android.rb

```diff
@@ -6,13 +6,16 @@ module Hive
   class Controller
     class Android < Controller
       attr_accessor :devices
+      attr_accessor :attached_devices
 
-      def detect
-        all_attached_devices = get_connected_devices # get all connected devices
-        to_poll = select_devices(all_attached_devices) # select devices to poll
+      def detect(device_type = 'Mobile')
+        # device_type should be either 'Mobile' or 'Tv'
+        @device_type = device_type
```

Comment:
Do we need to have this as an instance variable? It feels like a waste when we're only calling it once and setting it each time the detect method is called. Shouldn't get_connected_devices have a parameter?